snuba_sdk/delete_query.py

```diff
@@ -0,0 +1,50 @@
+import json
+from dataclasses import dataclass
+from typing import Any, Dict, List, Union
+
+from snuba_sdk.query import BaseQuery
+
+
+class InvalidDeleteQueryError(Exception):
+    pass
+
+
+@dataclass(frozen=True)
+class DeleteQuery(BaseQuery):
+    """
+    This represents a snuba delete query.
+    Inputs:
+        storage - the storage to delete from
+        columnConditions - a mapping from column-name to a list of column values
+            that defines the delete conditions. ex:
+            {
+                "id": [1, 2, 3]
+                "status": ["failed"]
+            }
+            represents
+            DELETE FROM ... WHERE id in (1,2,3) AND status='failed'
+    Deletes all rows in the given storage, that satisfy the conditions
+    defined in 'columnConditions'.
+    """
+
+    storage_name: str
+    column_conditions: Dict[str, List[Union[str, int]]]
+
+    def validate(self) -> None:
+        if "project_id" not in self.column_conditions:
```

Comment:
I was actually thinking of a validation that was more general, something like the following

```python
if self.column_conditions == {}:
    raise InvalidDeleteQueryError("column conditions cannot be empty")

for col, values in self.column_conditions.items():
    if len(values) == 0:
        raise InvalidDeleteQueryError(f"column condition {col} cannot be empty")
```

thoughts? that way we wouldn't have to update this code for every column we add